### PR TITLE
Sign up

### PR DESCRIPTION
**Pull request by afmicc**
#### Description

* Rename SignInForm component
* Add SignUp page
* Add ConfirmationSent page
* Add `useForm` hook to manage form

Screen Shot 2020-03-06 at 3 24 23 PM
Screen Shot 2020-03-06 at 3 25 06 PM

### CODE REVIEW

File: src/components/privateRoute/PrivateRoute.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ const PrivateRoute = ({ component: Component, ...props }) => {
 };
 
 PrivateRoute.propTypes = {
-  component: PropTypes.node.isRequired
+  component: PropTypes.func.isRequired
```
Comment (peter7z) [inline src/components/privateRoute/PrivateRoute.js]: You can destructure the propTypes right in the import like this:
```js
import { func } from 'prop-types';
```
And then:
```suggestion
  component: func.isRequired
```

File: src/components/signUpForm/SignUpForm.js [hunk 2]
```diff
@@ -1,29 +1,37 @@
...
 
-  const validField = (key, value) => !!value;
-  const validForm = validation.email && validation.password;
+  const dispatch = useDispatch();
+  const error = useSelector(({ user: { error } }) => error);
 
   const handleFormSubmit = event => {
     event.preventDefault();
-    if (validForm) dispatch(userActions.signIn(user));
+    if (validForm) dispatch(userActions.signUp(data));
```
Comment (peter7z) [inline src/components/signUpForm/SignUpForm.js]: Same here, you can import just the actions you are using.
```js
import { signUp } from 'redux/actions/userActions';
```

File: src/components/signUpForm/SignUpForm.js [hunk 3]
```diff
@@ -1,29 +1,37 @@
...
-import Button from 'components/common/button/Button';
+import FormSelect from 'components/common/formSelect/FormSelect';
+import useForm from 'hooks/useForm';
 
-const SignUpForm = () => {
-  const [user, setUser] = useState({ email: '', password: '' });
-  const [validation, setValidation] = useState({ email: undefined, password: undefined });
-  const error = useSelector(state => state.user.error);
-  const dispatch = useDispatch();
+const rules = {
```
Comment (peter7z) [inline src/components/signUpForm/SignUpForm.js]: This is usually called validations. You could also move this into a validations folder or sth like that.

File: src/redux/actions/userActions.js [hunk 4]
```diff
@@ -5,3 +5,8 @@ export const signInUserSuccess = createAction('SIGN_IN_USER_SUCCESS');
 export const signInUserError = createAction('SIGN_IN_USER_ERROR');
 export const signIn = credentials =>
   createDispatcher(userApi.signIn, signInUserSuccess, signInUserError, credentials);
+
+export const signUpSuccess = createAction('SIGN_UP_USER_SUCCESS');
+export const signUpError = createAction('SIGN_UP_USER_ERROR');
+export const signUp = params =>
```
Comment (peter7z) [inline src/redux/actions/userActions.js:11]: I really like this abstraction!
I think you are gonna love [redux-tools](https://github.com/rootstrap/redux-tools)

**Review by peter7z — APPROVED** [review]
Very nice!

File: src/components/signUpForm/SignUpForm.js (hunk 3 above)
Comment (afmicc) [reply src/components/signUpForm/SignUpForm.js]: As we talked, I'll call `config`

**Pull request merged**